ui/src/app/machines/components/TakeActionMenu/TakeActionMenu.tsx

```diff
@@ -1,93 +1,141 @@
-import { useEffect } from "react";
-
 import { ContextualMenu, Tooltip } from "@canonical/react-components";
+import type { ButtonProps } from "@canonical/react-components";
 import PropTypes from "prop-types";
-import { useDispatch, useSelector } from "react-redux";
+import { useSelector } from "react-redux";
 
 import type { MachineSetSelectedAction } from "app/machines/views/types";
-import { actions as generalActions } from "app/store/general";
-import { machineActions as machineActionsSelectors } from "app/store/general/selectors";
 import type { MachineAction } from "app/store/general/types";
 import machineSelectors from "app/store/machine/selectors";
 import type { Machine } from "app/store/machine/types";
 import { NodeActions } from "app/store/types/node";
 
+type ActionGroup = {
+  name: string;
+  actions: {
+    label: string;
+    name: MachineAction["name"];
+  }[];
+};
+
+type ActionLink = ButtonProps<{ "data-test"?: string }>;
```

Comment:
I'm suprised this is needed as ButtonProps looks like it is an intersection with `ButtonHTMLAttributes<HTMLButtonElement>`.

Reply:
I think the problem is the union with the generic `P`, i.e. `Omit<ButtonHTMLAttributes<HTMLButtonElement>, "onClick"> | P`, since for the type to be valid it needs to exist on both. Not sure how to fix this but I'll file a bug on react-components